skbio/sequence/tests/test_sequence.py

```diff
@@ -85,6 +85,11 @@ def test_concat_on_subclass(self):
         self.assertIs(type(result), SequenceSubclass)
         self.assertEqual(result, SequenceSubclass("123123"))
 
+    def test_concat_on_empty_iterator(self):
+        result = SequenceSubclass.concat((_ for _ in []))
```

Comment:
`numpy.concatenate` raises a `ValueError` in this case. Should we match that behavior?